apps/gateway/test/champions_test.exs

```diff
@@ -295,6 +297,105 @@ defmodule Gateway.Test.Champions do
+
+      {:ok, epic_bow} =
+        Items.insert_item_template(%{
+          game_id: Utils.game_id(),
+          name: "Epic Bow of Testness",
+          type: "weapon"
+        })
+
+      {:ok, item} = Items.insert_item(%{user_id: user.id, template_id: epic_bow.id, level: 1})
+
+      :ok = SocketTester.get_item(socket_tester, user.id, item.id)
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:item, %Item{} = fetched_item}
+      } = get_last_message()
+
+      assert fetched_item.id == item.id
+      assert fetched_item.user_id == user.id
+      assert fetched_item.unit_id == ""
+    end
+
+    test "equip and unequip item", %{socket_tester: socket_tester} do
+      # Register user
+      {:ok, user} = Users.register("EquipItemUser")
+
+      unit = user.units |> Enum.at(0)
+
+      {:ok, epic_sword} =
+        Items.insert_item_template(%{
+          game_id: Utils.game_id(),
+          name: "Epic Sword of Testness",
+          type: "weapon"
+        })
+
+      {:ok, item} = Items.insert_item(%{user_id: user.id, template_id: epic_sword.id, level: 1})
+
+      # EquipItem
+      :ok = SocketTester.equip_item(socket_tester, user.id, item.id, unit.id)
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:item, %Item{} = equipped_item}
+      } = get_last_message()
+
+      # The item is now equipped to the unit
+      assert equipped_item.user_id == user.id
+      assert equipped_item.unit_id == unit.id
+
+      # UnequipItem
+      :ok = SocketTester.unequip_item(socket_tester, user.id, item.id)
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:item, %Item{} = unequipped_item}
+      } = get_last_message()
+
+      # The item is now unequipped
+      assert unequipped_item.user_id == user.id
+      assert unequipped_item.unit_id == ""
+    end
+
+    test "level up item", %{socket_tester: socket_tester} do
+      # Register user
+      {:ok, user} = Users.register("LevelUpItemUser")
+
+      {:ok, epic_axe} =
+        Items.insert_item_template(%{
+          game_id: Utils.game_id(),
+          name: "Epic Axe of Testness",
+          type: "weapon"
+        })
+
+      {:ok, item} = Items.insert_item(%{user_id: user.id, template_id: epic_axe.id, level: 1})
+
+      gold_amount_before_level_up = Currencies.get_amount_of_currency_by_name(user.id, "Gold")
+
+      # LevelUpItem
+      :ok = SocketTester.level_up_item(socket_tester, user.id, item.id)
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:item, %Item{} = leveled_up_item}
+      } = get_last_message()
+
+      assert leveled_up_item.level == item.level + 1
+      assert Currencies.get_amount_of_currency_by_name(user.id, "Gold") < gold_amount_before_level_up
+
+      # Can not level up item if currency is not enough
+      level_up_item_until_unaffordable(socket_tester, user.id, item.id)
```

Comment:
I think it would be better if we set up our user to only have enough currency to upgrade once